### ISSUE TITLE
Is it possible to provide any context in the prompt

### ISSUE THREAD

**Issue by maziyarpanahi** [databrickslabs/dolly#55]
Hi,

Out of curiosity, is there any way to construct the PROMPT with some context to make sure the response is generated from that? Something similar to this discussion: https://community.openai.com/t/how-to-prevent-chatgpt-from-answering-questions-that-are-outside-the-scope-of-the-provided-context-in-the-system-role-message/112027

My use case is to first take the query and find top N semantically close documents from a database, then use them as context before pairing it with the input for a response. (a great way to make sure the generated response is not way off topic)

In alpaca, I saw this prompt:

```
f"""Below is an instruction that describes a task, paired with an input that provides further context. Write a response that appropriately completes the request.
# Instruction:
{instruction}
# Input:
{input}
# Response:
"""
```

**Comment by matthayes** [comment]
Yes you can provide context like below as in the Alpaca data. I would include the empty lines in between as this is how the training data looks. Also note there are three # in the training data. This also may be important.

```
Below is an instruction that describes a task, paired with an input that provides further context. Write a response that appropriately completes the request.

### Instruction:
{instruction}

### Input:
{input}

### Response:
```

**Comment by maziyarpanahi** [comment]
Thanks @matthayes 

I think it's working this way! I'll do more tests and if I can figure out how to do inline citation from that context then it's gold! 🥇 

Many thanks again

<img width="995" alt="image" src="https://user-images.githubusercontent.com/5762953/231248455-e3b88631-c189-4680-a906-9e405711f13a.png">

**Comment by opyate** [comment]
Just a head's up for anyone seeing this today, it seems the input key now doesn't have the `###` prepended anymore:
https://github.com/databrickslabs/dolly/blob/master/training/consts.py#L12
